src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -311,6 +311,142 @@ def k8s_install_cli(client_version='latest', install_location=None):
+    """Deploy the ACI-Connector to a AKS cluster
+    :param name: The name of the AKS cluster. The name is case insensitive.
+    :type name: str
+    :param connector_name: The name for the ACI Connector
+    :type connector_name: str
+    :param resource_group: Name of the resource group where the AKS is located and where
+    the ACI connector will be mapped.
+    :type resource_group: str
+    :param location:
+    :type location: str
+    :param service_principal: The service principal used for ACI authentication
+    to Azure APIs. If not specified, it is created for you and stored in the
+    ${HOME}/.azure directory.
+    :type service_principal: str
+    :param client_secret: The secret associated with the service principal. If
+    --service-principal is specified, then secret should also be specified. If
+    --service-principal is not specified, the secret is auto-generated for you
+    and stored in ${HOME}/.azure/ directory.
+    :type client_secret: str
+    :param chart_url: URL to the chart,
+    Default: https://github.com/Azure/aci-connector-k8s/raw/master/charts/aci-connector.tgz
+    :type chart_url: str
+    :param os_type: Os type target, could be Windows, Linux or Both.
+    :type os_type: str
+    """
+    from subprocess import PIPE, Popen
+    helm_not_installed = "Error : Helm not detected, please verify if it is installed."
+    image_tag = 'latest'
+    if chart_url is None:
+        url_chart = "https://github.com/Azure/aci-connector-k8s/raw/master/charts/aci-connector.tgz"
+    else:
+        url_chart = chart_url
+    # Check if Helm is installed locally
+    try:
+        Popen(["helm"], stdout=PIPE, stderr=PIPE)
+    except OSError:
+        raise CLIError(helm_not_installed)
+    # Validate if the RG exists
+    groups = _resource_client_factory().resource_groups
+    # Just do the get, we don't need the result, it will error out if the group doesn't exist.
+    rgkaci = groups.get(resource_group)
+    # Auto assign the location
+    if location is None:
+        location = rgkaci.location  # pylint:disable=no-member
+    # Get the credentials from a AKS instance
+    _, browse_path = tempfile.mkstemp()
+    aks_get_credentials(client, resource_group, name, admin=False, path=browse_path)
+    subscription_id = _get_subscription_id()
+    dns_name_prefix = _get_default_dns_prefix(connector_name, resource_group, subscription_id)
+    # Ensure that the SPN exists
+    principal_obj = _ensure_service_principal(service_principal, client_secret, subscription_id,
+                                              dns_name_prefix, location, connector_name)
+    client_secret = principal_obj.get("client_secret")
+    service_principal = principal_obj.get("service_principal")
+    # Get the TenantID
+    profile = Profile()
+    _, _, tenant_id = profile.get_login_credentials()
+    # Check if we want the windows connector
+    if (os_type == 'Windows') or (os_type == 'Both'):
```

Comment:
Define 'Windows', 'Both', 'Linux' as constants and this should be a case-insensitive comparison.